synfig-studio/src/gui/dialogs/dialog_setup.cpp

```diff
@@ -447,6 +457,69 @@ dragging the handle to the left bottom part of your 2D space."));
 	
 }
 
+void
+Dialog_Setup::create_shortcuts_page(Dialog_Template::PageInfo pi)
+{
+	attach_label_section(pi.grid, _("Keyboard Shortcuts"), 1);
+
+	Gtk::TreeModelColumnRecord columns;
+	Gtk::TreeModelColumn<std::string> action_name;
+	Gtk::TreeModelColumn<guint> action_key;
+	Gtk::TreeModelColumn<Gdk::ModifierType> action_mods;
+	columns.add(action_name);  //SHORTCUT_COLUMN_ID_ACTION_NAME
+	columns.add(action_key);   //SHORTCUT_COLUMN_ID_ACTION_KEY
+	columns.add(action_mods);  //SHORTCUT_COLUMN_ID_ACTION_MODS
+	auto model = Gtk::ListStore::create(columns);
+	treeview_accels = manage(new Gtk::TreeView(model));
+	treeview_accels->set_hexpand(true);
+	treeview_accels->set_vexpand(true);
+	treeview_accels->append_column("Action", action_name);
+
+	renderer_accel.property_editable() = true;
+
+	renderer_accel.signal_accel_edited().connect(sigc::mem_fun(*this, &Dialog_Setup::on_accel_edited));
+	renderer_accel.signal_accel_cleared().connect(sigc::mem_fun(*this, &Dialog_Setup::on_accel_cleared));
+
+	const int shortcut_col_idx = -1 + treeview_accels->append_column("Shortcut", renderer_accel);
+	treeview_accels->get_column(shortcut_col_idx)->add_attribute(renderer_accel, "accel-key", action_key);
+	treeview_accels->get_column(shortcut_col_idx)->add_attribute(renderer_accel, "accel-mods", action_mods);
+
+	auto map = App::get_default_accel_map();
+	for (const auto& pair : map) {
+		Gtk::AccelKey accel;
+		Gtk::TreeRow row = *model->append();
+
+		row.set_value(action_name, std::string(pair.second));
+
+		if (!Gtk::AccelMap::lookup_entry(pair.second, accel))
+			accel = Gtk::AccelKey(pair.first, pair.second);
+
+		row.set_value(action_key, accel.get_key());
+		row.set_value(action_mods, accel.get_mod());
+	}
+
+	Gtk::ScrolledWindow *scroll = manage(new Gtk::ScrolledWindow());
+	scroll->add(*treeview_accels);
+
+	pi.grid->attach(*scroll, 0, 1, 1, 1);
+}
+
+void
+Dialog_Setup::on_accel_edited(const Glib::ustring& path_string, guint accel_key, Gdk::ModifierType accel_mods, guint hardware_keycode)
+{
+	auto iter = treeview_accels->get_model()->get_iter(path_string);
+	iter->set_value(SHORTCUT_COLUMN_ID_ACTION_KEY, accel_key);
+	iter->set_value(SHORTCUT_COLUMN_ID_ACTION_MODS, accel_mods);
+	std::string action_name;
```

Comment:
Why is that needed?
```
std::string action_name;
iter->get_value(SHORTCUT_COLUMN_ID_ACTION_NAME, action_name);
```
Looks like it do nothing.

Reply:
Of course it does: it shows everyone I didn't properly clean my code ;)